packages/core/src/associations/belongs-to-many.ts

```diff
@@ -699,11 +699,27 @@ Add your own primary key to the through model, on different attributes than the
 
     const defaultAttributes = options?.through || {};
 
+    const multipleAttributes = Array.isArray(defaultAttributes);
+    const defaultAttributesArray: JoinTableAttributes[] = multipleAttributes ? defaultAttributes : [defaultAttributes];
+
+    // Check that there are the same new targets than attributes to match one to one
+    if (multipleAttributes && defaultAttributesArray.length !== newTargets.length) {
+      throw new AssociationError(`Error when adding or updating association between ${this.source.name} and ${newTargets.length} instances of ${this.target.name} when providing ${defaultAttributesArray.length} objects for the join table.
+      ${newTargets.length} and ${defaultAttributesArray.length} should be the same.
+      
+      Many to Many association join table extra attributes can be updated in bulk by passing an Array to the "through" parameter by doing this:    
```

Comment:
The rest of this message can be moved to the website. If they're already using this option they likely know what it is :)

Or we can merge the website documentation then link to it from here